soc/riscv/riscv-ite/it8xxx2/linker.ld

```diff
@@ -226,6 +239,21 @@ SECTIONS
     SECTION_PROLOGUE(ilm_ram,(NOLOAD),ALIGN(0x1000))
 	{
 		__ilm_ram_start = .;
+
+#ifdef CONFIG_SOC_IT8XXX2_SHA256_HW_ACCELERATE
+		__sha256_ram_block_start = .;
+		KEEP(*(.__sha256_ram_block))
```

Comment:
Same issue here - no code is using the `__sha256_ram_block` section.

Reply:
In WIP CL, we put message, hash and key data in __sha256_ram_block.

Reply:
Perhaps the hardware accelerator support should be implemented here as a crypto driver. If symbols are added to zephyr that aren't used by zephyr it seems likely that they could be removed later- better if the driver lives in the same repository so it's clearer what this is needed for.

Reply:
Ack

Reply:
+1 on moving the sha256 HW accelerator code upstream.  But approving this PR to unblock the downstream work.